### PR TITLE
Better documentation about Lizzy

### LINKED ISSUE #13: Better documentation
Hi!

I have no idea how to use this tool. So please provide better documentation and also some examples. I'm specially interested in using it within Continuous Delivery.

Thanks!

### PR DESCRIPTION
**Pull request by rafaelcaricio**
Fixes #13 

I have almost completely rewritten the README of Lizzy and I would like you to review and check for possible spelling errors or critics.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,64 +1,154 @@
...
 
+=====
 Lizzy
 =====
 
-REST Service to deploy AWS Cloud Formation templates using `Senza`_
-CLI tool.
+Lizzy is a wrapper around `Senza`_ (a command line deployment tool to
+create and execute AWS CloudFormation templates) that allows cross
+AWS deployments.
```
Comment (jmcs) [inline README.rst]: "cross AWS account"

File: README.rst [hunk 2]
```diff
@@ -1,64 +1,241 @@
...
+Why Lizzy?
+==========
+
+Lizzy was created with the objective of enabling autonomous teams at
+`Zalando`_ to use Jenkins pipelines to continuously deploy new versions of
+their apps to their AWS accounts using the `Senza`_ CLI tool.
+
+The managed Jenkinses at Zalando run under the Continuous Delivery team's
+AWS account. The Continuous Delivery team is responsible for
+maintaining the Jenkinses of all other Zalando teams. That allows
```
Comment (last-ent) [inline README.rst]: ... teams. `This` allows

File: README.rst [hunk 3]
```diff
@@ -1,64 +1,241 @@
...
+other teams in the company to focus on their `OKR's`_ and not invest
+their time on configuring Jenkins or continuous deployment tools. The
+teams at Zalando enjoy `Jenkins-as-a-Service`_.
+
+Each team at Zalando has their own AWS account. So running the
+`Senza`_ tool within a managed Jenkins job would allow deployments only in
+the Continuous Delivery team's AWS account. To permit different teams
```
Comment (last-ent) [inline README.rst]: > So running the `Senza`_ tool within a managed Jenkins job would allow deployments only in the Continuous Delivery team's AWS account.

It's not clear to me what you are trying to convey here, I am afraid. 
Do you mean to say that Senza tool runs managed Jenkins jobs in CD team's AWS account?

File: README.rst [hunk 4]
```diff
@@ -1,64 +1,241 @@
...
+to deploy in their own AWS account Lizzy was created.
+
+How Lizzy works?
+================
+
+Lizzy consists of a REST API service (also called Agent) and the
+`Lizzy-Client`_ tool.
+
+The Lizzy Agent is a web application that is deployed in the team's
```
Comment (last-ent) [inline README.rst]: > in the team's account

Does the `team` refer to CD Team or the team who is using Senza?

File: README.rst (hunk 3 above)
Comment (rafaelcaricio) [inline README.rst]: No, I am trying to say that if you run `senza` tool in a managed Jenkins job it will try to deploy to the Continuous Delivery team AWS account.

Comment (rafaelcaricio) [inline README.rst]: I will try rewording that.

File: README.rst (hunk 4 above)
Comment (rafaelcaricio) [inline README.rst]: The team that is using Senza.

File: README.rst [hunk 5]
```diff
@@ -1,64 +1,241 @@
...
+
+    $ docker build -t lizzy:dev .
+
+
+.. hint:: The "scm-source.json" file will be missing if you just clone
+          this repository. To know more what is this file, please read
+          the `STUPS documentation`_. We have `tools`_ that can generate
+          this file for you.
+
+After the image build, it will be available within the tag
```
Comment (last-ent) [inline README.rst]: Slight modifications here:
*After the image `is built`, it will be available `with` the tag...*

File: README.rst [hunk 6]
```diff
@@ -1,64 +1,241 @@
...
+your machine under `~/.aws`. After that you can run the Lizzy image
+with the command:
 
+.. code-block:: sh
+
+    $ docker run --rm -p 8080:8080 --env-file environment.cfg -v ~/.aws:/.aws --name lizzy -u 999 -it lizzy:dev
+
+The application by default will be listening on port `8080`. Usually
+accessible at `http://127.0.0.1:8080`, it depends on how your Docker is
+configured locally. A Swagger/OpenAPI console will be available at
```
Comment (last-ent) [inline README.rst]: Do you mean to say
> By default, the application will be accessible on port `8080` & on url `http://127.0.0.1/` though this might vary based on how Docker is configured on the local system?

File: README.rst [hunk 7]
```diff
@@ -1,64 +1,241 @@
...
+          HTTPPort: 8080
+          HealthCheckPath: /api/swagger.json
+          Scheme: internet-facing
+          SecurityGroups: ['app-lizzy-lb']
+          Type: Senza::WeightedDnsElasticLoadBalancer
+
+Access Control for Lizzy
+------------------------
+
+Lizzy application will need to have access to create new Cloud
```
Comment (last-ent) [inline README.rst]: I feel the sentence is slightly off and it should be
> Lizzy application should have access to create new...

the use two `to` in close proximity feels awkward. I might be mistaken here though.

File: README.rst [hunk 8]
```diff
@@ -1,64 +1,241 @@
...
+Formation stacks and some others services from Amazon API. You will
+need to specify the a `IAM role`_ similar to:
```
Comment (last-ent) [inline README.rst]: > specify the `IAM role`

File: README.rst [hunk 9]
```diff
@@ -1,64 +1,241 @@
...
+                    "autoscaling:*"
+                ],
+                "Effect": "Allow",
+                "Resource": "*"
+            }
+        ],
+        "Version": "2012-10-17"
+    }
+
+That is the minimal configuration needed for Lizzy to be able to run
```
Comment (last-ent) [inline README.rst]: `This` is the minimal...

File: README.rst [hunk 10]
```diff
@@ -1,64 +1,241 @@
...
+successfully Senza commands. Other statements might be included in this
```
Comment (last-ent) [inline README.rst]: The word placement is off.
> This is the minimal configuration needed for Lizzty to run Senza command successfully.

Comment (last-ent) [inline README.rst]: > Other statements might be included in `this`
Use of `this` here is too vague. 

Also, I don't think it is specified where the above stated json file is saved and by what name.

Comment (rafaelcaricio) [inline README.rst]: It is the AIM role, that is why there is a link to amazon documentation.

Comment (last-ent) [inline README.rst]: Ah ok. Then I am not sure what the statement  `Other statements might be included in this`  means.

Comment (rafaelcaricio) [inline README.rst]: Actually should be "Other statements might be included in this IAM role."

Comment (last-ent) [inline README.rst]: That makes sense.

**Comment by LappleApple** [comment]
Done (HipChat not working, want to make sure you see this)

**Comment by rafaelcaricio** [comment]
@LappleApple Thank you! 💕 I will read everything again and fix the formatting.

**Comment by rafaelcaricio** [comment]
👍

**Comment by LappleApple** [comment]
Just one question @rafaelcaricio @jmcs: Does this mean that Lizzy is now more "out-of-the-box" useful, right? Can someone use Lizzy, along with Senza, and have zero dependency issues on other-STUPS/anything else? If so, we can state that very clearly at the top, so folks are 100% sure. And we can also highlight the Docker file in the intro, too, to draw attention to it right away.

**Comment by rafaelcaricio** [comment]
@LappleApple It is possible to use Lizzy (&& Senza) without any other STUPS infrastructure.

Just the problem that Lizzy solves is very tight to the use case of how we work at Zalando currently, as described in the section ["Why Lizzy Exists"](https://github.com/zalando/lizzy#why-lizzy-exists).

It might be useful for teams that already use Senza and want to adopt a continuous delivery approach.

Lizzy (neither Lizzy-Client) are neither tied to Jenkins. It can be used with any continuous delivery tool like Travis, CircleCI, GoCD, etc.

**Comment by LappleApple** [comment]
Thanks, @rafaelcaricio. To broaden Lizzy's appeal, I would rephrase "Why Lizzy Exists" by adding at least one sentence that would make Lizzy interesting to others. Other companies work similarly, either by providing AWS accounts to indiv teams or doing Jenkins as a Service or both. So a sentence to the effect of, "If you're on a team that ..." can make it clear why someone would wan to use it (and Senza). You can acknowledge that Lizzy is very opinionated to our needs but use a sentence try to speak to others who have those same needs.

I would also point out that you can use Lizzy with any other CD tool.

**Comment by jkozlowicz** [comment]
👍

**Comment by rafaelcaricio** [comment]
@LappleApple Added to the readme. 👍
